plugins/modules/site_workflow_manager.py

```diff
@@ -981,29 +1787,70 @@ def verify_diff_merged(self, config):
             (have) and desired state (want) of the configuration, logs the states, and validates whether the specified
             site exists in the Catalyst Center configuration.
         """
+        try:
+            if self.get_ccc_version_as_integer() >= self.version_2_3_7_6 and self.config[0]['site']['type'] == "bulk":
+                name_list = []
+                for site in self.want:
+                    if "name" in site:
+                        name_list.append(site["name"])
+                self.get_have(config)
+                site_exist = self.have.get("site_exists")
+                # self.log(site_exist)
+                if site_exist:
+                    self.status = "success"
+                    self.msg = "The requested site '{0}' is present in the Cisco Catalyst Center and its creation has been verified.".format(
+                        name_list)
+                    self.log(self.msg, "INFO")
+                return self
+
+        except Exception as e:
+            self.log("An unexpected error occurred: {0}".format(e), "ERROR")
+
+        self.get_have(config)
+        self.log("Current State (have): {0}".format(str(self.have)), "INFO")
+        self.log("Desired State (want): {0}".format(str(self.want)), "INFO")
+        site_exist = self.have.get("site_exists")
+        site_name = self.want.get("site_name")
```

Comment:
Call the get_want and get_have in the beginning of API only. Don't call it multiple times as it slow the execution because the purpose is to check we got the desired state or not after operation.

Reply:
Removed